ci-operator/step-registry/gather/core-dump/gather-core-dump-commands.sh

```diff
@@ -12,5 +12,36 @@ echo "Gathering node core dumps ..."
 mkdir -p ${ARTIFACT_DIR}/core-dumps
 
 oc adm must-gather --dest-dir="${ARTIFACT_DIR}/core-dumps" -- sh -c "/usr/bin/gather_core_dumps || true"
+
+ls -R ${ARTIFACT_DIR}/core-dumps
+find ${ARTIFACT_DIR}/core-dumps/*/ -type f
+CORE_DUMPS="$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f)"
+num_core_dumps="$(echo -n ${CORE_DUMPS} | grep -c "^" || true)"
```

Comment:
The first two lines seem essentially equivalent? Also, the `/*/` in the find command is superfluous isn't it? (Well, I guess if there are files and not just subdirectories in `${ARTIFACT_DIR}/core-dumps/` then it skips them?)

Also, `grep -c "^"` is more idiomatically expressed as `wc -l`, but at any rate, won't the answer always be `1` because the newlines in the `find` output will have been turned into spaces by `$()`?

Reply:
yeah, the ls and find lines are similar. I can remove the ls if you think it's too noisy.

here's an [example log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-master-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade/1503943499051962368/artifacts/e2e-aws-ovn-upgrade/gather-core-dump/build-log.txt).

it took me a bit to get the grep/wc stuff correct, but it works like this. here's it [passing](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-release-4.9-4.9-upgrade-from-stable-4.8-e2e-aws-ovn-upgrade/1503847329612959744/artifacts/e2e-aws-ovn-upgrade/gather-core-dump/build-log.txt) (no core files) and [failing](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-master-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade/1503943499051962368/artifacts/e2e-aws-ovn-upgrade/gather-core-dump/build-log.txt).

```
❯ echo $CORE_DUMPS

❯ num_core_dumps="$(echo -n ${CORE_DUMPS} | grep -c "^" || true)"
❯ echo $num_core_dumps
0

```
and
```
❯ CORE_DUMPS="$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f)"
❯ echo $CORE_DUMPS
/tmp/core-dumps/node1/jamo.lz1
❯ num_core_dumps="$(echo -n ${CORE_DUMPS} | grep -c "^" || true)"
❯ echo $num_core_dumps
1
```

Reply:
yeah, ok, it gets 0 vs 1 correct, but not 1 vs more than 1:

```
danw@p1:~> touch /tmp/core-dumps/node1/core
danw@p1:~> touch /tmp/core-dumps/node2/core
danw@p1:~> CORE_DUMPS="$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f)"
danw@p1:~> echo $CORE_DUMPS
/tmp/core-dumps/node1/core /tmp/core-dumps/node2/core
danw@p1:~> num_core_dumps="$(echo -n ${CORE_DUMPS} | grep -c "^" || true)"
danw@p1:~> echo $num_core_dumps
1
```

Reply:
gotcha. I am using zsh locally and this doesn't happen so didn't realize. don't need the quotes around var in echo to
output with newlines:

```
❯ CORE_DUMPS="$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f)"
❯ echo $CORE_DUMPS
/tmp/core-dumps/node1/jamo2.lz1
/tmp/core-dumps/node1/jamo.lz1
❯ num_core_dumps="$(echo -n ${CORE_DUMPS} | grep -c "^" || true)"
❯ echo $num_core_dumps
2
```
I just updated to use quotes. Also removed the extra 'ls' you pointed out is mostly just adding noise. need to get through one round of presubmits with 4.8->4.9 upgrade job happening to double check all is well. after that, I'll update again to remove 4.8->4.9 from this PR and hopefully then it's good to go.

/hold

Reply:
/hold cancel

alright @danwinship, I think this is right now.

- [job that passes](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade/1506332086284324864) and it's [core-dump build-log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade/1506332086284324864/artifacts/e2e-aws-ovn-upgrade/gather-core-dump/build-log.txt)
- job that fails [because of core dumps](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-release-4.9-4.9-upgrade-from-stable-4.8-e2e-aws-ovn-upgrade/1506332085386743808) and the build log showing it found [6 of them](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-release-4.9-4.9-upgrade-from-stable-4.8-e2e-aws-ovn-upgrade/1506332085386743808/artifacts/e2e-aws-ovn-upgrade/gather-core-dump/build-log.txt).
- finally just a [failing job](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-pull-ci-openshift-ovn-kubernetes-release-4.10-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade/1506332085642596352) for some of the normal reasons (but not core dump) we have upgrade failures.